src/main/java/com/cs3365/rpgdapi/weapons/WeaponsController.java

```diff
@@ -64,4 +59,40 @@ public ResponseEntity get(@PathVariable("weaponID") UUID identifier)
                                          e.getMessage()));
         }
     }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<String> put(@RequestBody Weapon requestBody,
+                                      @PathVariable UUID identity) throws WeaponsException {
+        try {
+            requestBody.validate();
```

Comment:
Does calling `.validate()` on the request body actually work? I usually write my own validation logic in a validator class or a request object class when doing this, so I've never seen this used before.

Reply:
It seems to work, but I'm honestly not too sure how. I just noticed that it was in the POST request so I did that.